src/hub_api.rs

```diff
@@ -36,6 +36,9 @@ pub struct IndexEntry {
     language: String,
     author: String,
     tags: Vec<String>,
+    url: String,
+    id: String,
```

Comment:
I left a comment on the design doc about understanding whether this was specifically a template ID (in which case not every entry will have it) or an entry ID (in which case every entry will have it, but it can't be guaranteed to be the same as the template ID).

If this is the template ID, which is certainly how it's used in this PR, then I'd strongly encourage you to call it `template_id` rather than plain `id`.